core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/UpxCompressionBuildStep.java

```diff
@@ -40,10 +40,15 @@ public void compress(NativeConfig nativeConfig, NativeImageRunnerBuildItem nativ
             NativeImageBuildItem image,
             BuildProducer<UpxCompressedBuildItem> upxCompressedProducer,
             BuildProducer<ArtifactResultBuildItem> artifactResultProducer) {
+
         if (nativeConfig.compression().level().isEmpty()) {
             log.debug("UPX compression disabled");
             return;
         }
+        if (image.isReused()) {
+            log.info("Native executable reused: skipping compression");
```

Comment:
I'm not entirely sure if we want an info line here. I would demote it to debug given it's really an implementation detail (I mean you for sure don't expect things to be compressed twice if you reuse an executable).

The why is that we are trying hard to avoid having a lot of log lines so that people don't miss anything important.

Do you agree? (Don't hesitate to push back if you don't) If so, either push and squash or I'll be happy to do it if you are not comfortable with git yet.

Reply:
This sounds good to me.
Done.